src/main/java/frc/robot/robot_manager/RobotManager.java

```diff
@@ -1,18 +1,207 @@
+          IDLE_NO_GP,
+          IDLE_WITH_GP,
+          CLIMBING_1_LINEUP,
+          CLIMBING_2_HANGING -> {}
+      case SPEAKER_SCORING -> {
+        if (!queuer.hasNote()) {
+          setStateFromRequest(RobotState.IDLE_NO_GP);
+        }
+      }
+      case AMP_SCORING -> {
+        if (!queuer.hasNote()) {
+          setStateFromRequest(RobotState.IDLE_NO_GP);
+        }
+      }
+      case FEEDING_SHOOTING -> {
+        if (!queuer.hasNote()) {
+          setStateFromRequest(RobotState.IDLE_NO_GP);
+        }
+      }
+      case PASS_SHOOTING -> {
+        if (!queuer.hasNote()) {
+          setStateFromRequest(RobotState.IDLE_NO_GP);
+        }
+      }
+      case SPEAKER_PREPARE_TO_SCORE -> {
+        if (shooter.atGoal() && arm.atGoal()) {
+          setStateFromRequest(RobotState.SPEAKER_SCORING);
+        }
+      }
+
+      case AMP_PREPARE_TO_SCORE -> {
+        if (shooter.atGoal() && arm.atGoal()) {
+          setStateFromRequest(RobotState.AMP_SCORING);
+        }
+      }
+      case FEEDING_PREPARE_TO_SHOOT -> {
+        if (shooter.atGoal() && arm.atGoal()) {
+          setStateFromRequest(RobotState.FEEDING_SHOOTING);
+        }
+      }
+      case PASS_PREPARE_TO_SHOOT -> {
+        if (shooter.atGoal() && arm.atGoal()) {
+          setStateFromRequest(RobotState.PASS_SHOOTING);
+        }
+      }
+      case UNJAM -> {}
+      case INTAKING -> {
+        if (intake.hasNote() || queuer.hasNote()) {
+          setStateFromRequest(RobotState.IDLE_WITH_GP);
+        }
+      }
+      case OUTTAKING -> {
+        if (!intake.hasNote() && !queuer.hasNote()) {
+          setStateFromRequest(RobotState.IDLE_NO_GP);
+        }
+      }
+    }
+
+    return currentState;
+  }
+
+  @Override
+  protected void afterTransition(RobotState newState) {
+    switch (newState) {
+      case SPEAKER_PREPARE_TO_SCORE, SPEAKER_SCORING, SPEAKER_WAITING -> {
+        arm.setState(ArmState.SPEAKER_SHOT);
+        shooter.setState(ShooterState.SPEAKER_SHOT);
+        intake.setState(IntakeState.IDLE);
+        queuer.seState(QueuerState.IDLE_WITH_GP);
+      }
+      case AMP_PREPARE_TO_SCORE, AMP_SCORING, AMP_WAITING -> {
+        arm.setState(ArmState.AMP);
+        shooter.setState(ShooterState.AMP);
+        intake.setState(IntakeState.IDLE);
+        queuer.seState(QueuerState.IDLE_WITH_GP);
+      }
+      case FEEDING_PREPARE_TO_SHOOT, FEEDING_SHOOTING, FEEDING_WAITING -> {
+        arm.setState(ArmState.FEEDING);
+        shooter.setState(ShooterState.FEEDING);
+        intake.setState(IntakeState.IDLE);
+        queuer.seState(QueuerState.IDLE_WITH_GP);
+      }
+      case PASS_PREPARE_TO_SHOOT, PASS_SHOOTING, PASS_WAITING -> {
+        arm.setState(ArmState.PASS);
+        shooter.setState(ShooterState.PASS);
+        intake.setState(IntakeState.IDLE);
+        queuer.seState(QueuerState.IDLE_WITH_GP);
+      }
+      case UNJAM -> {
+        arm.setState(ArmState.AMP);
+        shooter.setState(ShooterState.DROP);
```

Comment:
i think using the pass state here may be preferable, since it's a higher RPM and might help unjam better